vnext/ReactUWP/Modules/AsyncStorageModuleWin32.cpp

```diff
@@ -0,0 +1,191 @@
+#include "pch.h"
+
+#include "AsyncStorageModuleWin32.h"
+
+#include <cstdio>
+
+/// Implements ASM using winsqlite3.dll (requires Windows version 10.0.10586)
+
+namespace facebook {
+namespace react {
+
+namespace {
+
+static void ExecImpl(sqlite3 *db, const char *statement, AsyncStorageModuleWin32::ExecCallback callback, void *pv) {
+  char *errMsg = nullptr;
+  int rc = sqlite3_exec(db, statement, callback, pv, &errMsg);
+  if (errMsg) {
+    throw std::runtime_error(errMsg);
+  }
+  if (rc != SQLITE_OK) {
+    throw std::runtime_error(sqlite3_errmsg(db));
+  }
+}
+
+class Sqlite3Transaction {
+  sqlite3 *m_db{nullptr};
+
+ public:
+  explicit Sqlite3Transaction(sqlite3 *db) : m_db(db) {
+    if (db) {
+      ExecImpl(db, u8"BEGIN TRANSACTION", nullptr, nullptr);
+    }
+  }
+  Sqlite3Transaction(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction(Sqlite3Transaction &&other) : m_db{other.m_db} {
+    other.m_db = nullptr;
+  }
+  Sqlite3Transaction &operator=(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction &operator=(Sqlite3Transaction &&rhs) {
+    std::swap(m_db, rhs.m_db);
+  }
+  ~Sqlite3Transaction() {
+    ExecImpl(m_db, u8"COMMIT", nullptr, nullptr);
+  }
+};
+
+} // namespace
+
+AsyncStorageModuleWin32::AsyncStorageModuleWin32(PCSTR dbPath) {
+  checkSQLiteResult(
+      sqlite3_open_v2(dbPath, &m_db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FULLMUTEX, nullptr));
+
+  int userVersion = 0;
+  auto userVersionCallback = [&](int cCol, char **rgszColText, char **rgszColName) {
+    userVersion = atoi(rgszColText[0]);
+    return SQLITE_OK;
+  };
+
+  Exec(u8"PRAGMA user_version", userVersionCallback);
+
+  if (userVersion == 0) {
+    Exec(u8"CREATE TABLE IF NOT EXISTS AsyncLocalStorage(key TEXT, value TEXT); PRAGMA user_version=1");
+  }
+}
+
+AsyncStorageModuleWin32::~AsyncStorageModuleWin32() {
+  checkSQLiteResult(sqlite3_close(m_db));
+}
+
+std::string AsyncStorageModuleWin32::getName() {
+  return "AsyncLocalStorage";
+}
+
+std::map<std::string, dynamic> AsyncStorageModuleWin32::getConstants() {
+  return {};
+}
+
+std::vector<facebook::xplat::module::CxxModule::Method> AsyncStorageModuleWin32::getMethods() {
+  return {Method("multiGet", this, &AsyncStorageModuleWin32::multiGet),
+          Method("multiSet", this, &AsyncStorageModuleWin32::multiSet),
+          Method("multiRemove", this, &AsyncStorageModuleWin32::multiRemove),
+          // The 'multiMerge' method is currently not implemented. We assume that
+          // the ReactNative framework translates its absence into the AsyncStorage
+          // JS object (see Libraries\Storage\AsyncStorage.js around line 485) which
```

Comment:
This line no will change (especially since built-in Asyncstorage is going away). I would just omit it.